carbon-design-system/carbon-addons-iot-react
# feat(imagehotspots): add selecting and creating hotspots

Closes #1770

**Summary**
The ImageHotspots is enhanced to support selecting hotspots for editing and adding new ones by clicking on the image. It is **still possible to drag the image** when in isEditable mode which makes sense for precision positioning of new hotspots. 

**Change List (commits, features, bugs, etc)**
- Added isEditable prop to ImageHotspots.jsx inorder to enable selection and adding hotshpots
- Added callback for when a hotspot is clicked (selected) in the ImageHotspots.jsx
- Added callback for clicking the image to create a new hotspot ImageHotspots.jsx
- Changed the drag code to start drag on mouse move instead of mouse down in ImageHotspots.jsx
- Added isSelected to Hotspot.jsx
- Refactored the StyledComponents into external css for Hotspot.jsx
- Added a story
- Added unit tests

**Acceptance Test (how to verify the PR)**
1. Go to https://deploy-preview-1813--carbon-addons-iot-react.netlify.app?path=/story/watson-iot-imagehotspots--editable
2. Try selecting exiting hotspots. Make sure they don't move.
3. Try adding new hotspots by clicking on the map.
5. Zoom and make sure you can stil drag

**Note!** 
Clicking on hotspots doesn't always show the tooltip. This is an existing bug for which I filed a issue #1808
Apparently the same issue also prevents the selection from happening sometimes here. It works in dev mode so I assume it is a timing issue. I will look into that.
